### PR TITLE
Run tests on multiples environments (db versions)

### PR DESCRIPTION
**Pull request by slimee**

### CODE REVIEW

File: .env-test-max-version.ci [hunk 1]
```diff
@@ -0,0 +1,3 @@
+DEVOPS_SLACK_CHANNEL=
+DEVOPS_SLACK_TOKEN=
```
Comment (VincentMolinie) [inline .env-test-max-version.ci]: [MISC] What is this doing here ?

File: .env-test-min-version.ci [hunk 2]
```diff
@@ -0,0 +1,3 @@
+DEVOPS_SLACK_CHANNEL=
+DEVOPS_SLACK_TOKEN=
```
Comment (VincentMolinie) [inline .env-test-min-version.ci]: [MISC] What is this doing here ?

File: .env-test-max-version.ci [hunk 3]
```diff
@@ -0,0 +1,3 @@
...
+MONGO_URL=mongodb://localhost:27016
```
Comment (VincentMolinie) [inline .env-test-max-version.ci]: [Code Convention] Missing line return at EOF

File: .travis.yml [hunk 4]
```diff
@@ -16,11 +16,11 @@ services:
 before_install:
   - docker-compose --version
   - docker-compose up -d
-  - docker ps
+  - docker ps -a
```
Comment (VincentMolinie) [inline .travis.yml]: [MISC] Why ?

File: package.json [hunk 5]
```diff
@@ -6,6 +6,7 @@
   "scripts": {
     "lint": "./node_modules/eslint/bin/eslint.js ./*.js ./assets ./deserializers ./serializers ./services ./test",
     "test": "./node_modules/mocha/bin/mocha 'test/**/*.js'",
```
Comment (VincentMolinie) [inline package.json]: [Specs] The yarn test command seems to be broken 🤔

File: package.json [hunk 6]
```diff
@@ -6,6 +6,7 @@
...
+    "test:ci": "ENV=.env-test-min-version.ci npm run test && ENV=.env-test-max-version.ci npm run test",
```
Comment (VincentMolinie) [inline package.json]: [Specs] There is something that I really don't like in this solution is that not all the test needs to have different environment. Why not just having multiple environment variables and in the test do something like a foreach (Or an helper to do it) Like that we will have the context in the `describe` and  we don't have to run twice the test for the tests that does not need it

File: .travis.yml (hunk 4 above)
Comment (slimee) [reply .travis.yml]: Good to debug when a container is nok

File: package.json (hunk 5 above)
Comment (slimee) [reply package.json]: What was the problem?

File: package.json (hunk 6 above)
Comment (slimee) [reply package.json]: That' a good point indeed.

Comment (slimee) [reply package.json]: We will think about it before continue 👍

Comment (slimee) [reply package.json]: I've just produced a new version where tests are categorized and better selected 👍

File: package.json [hunk 7]
```diff
@@ -6,6 +6,8 @@
   "scripts": {
     "lint": "./node_modules/eslint/bin/eslint.js ./*.js ./assets ./deserializers ./serializers ./services ./test",
     "test": "./node_modules/mocha/bin/mocha 'test/**/*.js'",
+    "test:int": "./node_modules/mocha/bin/mocha 'test/integration/**/*.js'",
+    "test:ci": "ENV=.env-test npm run test && ENV=.env-test-old-db npm run test:int",
```
Comment (VincentMolinie) [inline package.json]: I still really don't like that and it's harder to know on which database you are. And if you want to add one  more version you need to change this command line every time.
We could do something like that: https://github.com/ForestAdmin/lumber/pull/318
It does a simple describe and it's pretty easy to use/extends
Capture d’écran 2019-11-07 à 11 19 56

**Pull request closed without merging**